commit c13ace9a9d0f2b0a2d5776f083cfdec62b6df756
Author: Volker Krause <vkrause@kde.org>
Date:   Sat Sep 26 21:42:08 2026 +0200

    Fix build with Qt 6.11.2 on Android
    
    That enables large file support by default, changing what QT_STATBUF
    defines to and thus no longer matching what the non-64 fstatat call
    expects.

diff --git a/src/kioworkers/file/stat_unix.h b/src/kioworkers/file/stat_unix.h
index f85f838f7..3ce019ec8 100644
--- a/src/kioworkers/file/stat_unix.h
+++ b/src/kioworkers/file/stat_unix.h
@@ -166,7 +166,11 @@ inline static uint64_t stat_mnt_id(const struct statx &buf)
 inline int LSTATAT(int dfd, const char *path, QT_STATBUF *buff, KIO::StatDetails details)
 {
     Q_UNUSED(details)
+#if defined(QT_LARGEFILE_SUPPORT) && defined(Q_OS_ANDROID)
+    return fstatat64(dfd, path, buff, AT_SYMLINK_NOFOLLOW);
+#else
     return fstatat(dfd, path, buff, AT_SYMLINK_NOFOLLOW);
+#endif
 }
 #endif
 
